### PR TITLE
Fix failing court benchmark

### PR DESCRIPTION
**Pull request by Chralt98**
### What does it do?

Fixes main branch CI https://github.com/zeitgeistpm/zeitgeist/actions/runs/6848052314/job/18617476375

### CODE REVIEW

**Comment by maltekliemann** [comment]
What exactly happened here? Why wasn't this caught by the CI?

**Comment by Chralt98** [comment]
> What exactly happened here? Why wasn't this caught by the CI?

It was caught by the CI, but because we usually expect the `No space left on device` error, it accidentally wasn't caught.

After integrating verify checks for court, one verify check was only visible for the actual execution of the benchmarking logic and couldn't be caught by the benchmark test suite. I don't know exactly why this happened, maybe because we use a mock and therefore also mock our auto resolve dispute api :D.

**Comment by maltekliemann** [comment]
Yeah, the `DisputeResolutionApi` mock incorrectly mirrors the behavior of the implementation in prediction-markets. The former removes _all_ occurences of the market ID while the latter only appears to remove the first item. Could you change the mock real quick so this sort of problem doesn't occur again?

**Comment by Chralt98** [comment]
2d2c39dae1cd43efbdbcab78ca08ea6987bf6f9f

**Comment by Chralt98** [comment]
I think this example emphasises the strong need of a general mock with all the dependencies as in the actual runtime. By putting in the whole prediction markets pallet configuration into the court mock, we could have prevented my "current mirroring change".

**Comment by Chralt98** [comment]
https://github.com/zeitgeistpm/zeitgeist/issues/1086 We could have prevented this mistake by using either a full mock of all pallets or the runtime itself as a test runtime.

File: zrml/court/src/mock.rs [hunk 1]
```diff
@@ -127,8 +127,11 @@ impl DisputeResolutionApi for MockResolution {
 
     fn remove_auto_resolve(market_id: &Self::MarketId, resolve_at: Self::BlockNumber) -> u32 {
         <mock_storage::MarketIdsPerDisputeBlock<Runtime>>::mutate(resolve_at, |ids| -> u32 {
-            ids.retain(|id| id != market_id);
-            ids.len() as u32
+            let ids_len = ids.len() as u32;
+            if let Some(pos) = ids.iter().position(|i| i == market_id) {
+                ids.swap_remove(pos);
```
Comment (maltekliemann) [inline zrml/court/src/mock.rs:132]:
```suggestion
                let _ = ids.swap_remove(pos);
```

**Review by maltekliemann — APPROVED** [review]

**Pull request merged**